### PR TITLE
feat: Introduce Rankings for Items, Creators & Collectors

### LINKED ISSUE #632: Rankings Section (Table)
The rankings section will show information based on a couple of filters. For this first issue, we'll need to only draw the table and filters, without any functionality.

To complete this issue:
- [ ] Show the rankings section table with a dynamic set of columns. Remember the total volume tooltip
- [ ] Add the type: Wearables / Creators / Collectors
- [ ] Add the Dropdown filter. By Volume or Quantity of items
- [ ] Add the time filters
- [ ] Only for the wearables view (so it should be hideable somehow) add the categories/rarities filters

### LINKED ISSUE #633: Rankings (Wearables Section)
Add all the information and filters for the wearables Rankings Section

to complete this feature:
- [ ] Fetch the wearable rankings. It should come pre-filtered by the volume/time filters, [tackled here](#634).
- [ ] Show the data in the table
- [ ] Implement the categories filter (keep in mind it should work with the other filters)
- [ ] Implement the rarities filter (keep in mind it should work with the other filters)

### LINKED ISSUE #634: Rankings Section (Header filter)
Implement the two main filters for the rankings table. They should affect all categories to show.

To complete this issue:
- Dropdown
  - [ ] Implement a MANA volume filter (on by default)
  - [ ] Implement a qty of items filter
- Time
  - [ ] Filter by the three different time values (last 7 days by default)

Both filters should work together

### LINKED ISSUE #635: Rankings (Creators Section)
Add all the information and filters for the Creators Rankings Section.

To complete this issue:
- [ ] Fetch the creators ranking using the [active filters](#634)
- [ ] Show the data on the table

### LINKED ISSUE #636: Rankings (Collectors Section)
Add all the information and filters for the Collectors Rankings Section.

To complete this issue:
- [ ] Fetch the collectors ranking using the [active filters](#634)
- [ ] Show the data on the table

### PR DESCRIPTION
**Pull request by juanmahidalgo**
Closes #632
Closes #633
Closes #634
Closes #635
Closes #636 

https://user-images.githubusercontent.com/8763687/172620036-afea618c-d7ec-4137-8130-62ab84c9ce83.mov

### CODE REVIEW

File: webapp/src/components/RankingsTable/RankingsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+              contractAddress={entity.id.split('-')[0]}
+              tokenId={entity.id.split('-')[1]}
+            >
+              {(item, order, isLoading) => {
+                if (!isLoading && !item) {
+                  return null
+                }
+                return (
+                  <Table.Row>
+                    <>
```
Comment (nicosantangelo) [inline webapp/src/components/RankingsTable/RankingsTable.tsx:214]: is this React.Fragment needed?

File: webapp/src/components/RankingsTable/RankingsTable.tsx [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+            onChange={timeframe => setCurrentTimeframe(timeframe)}
+          />
+        </HeaderMenu.Right>
+      </HeaderMenu>
+      <div className="rankings-card">
+        {renderTableTabs()}
+        <Loader active={isLoading} size="large" />
+
+        {isLoading ? (
+          <Loader active size="large" />
```
Comment (nicosantangelo) [inline webapp/src/components/RankingsTable/RankingsTable.tsx:419]: wouldn't this create two loaders? or am I tripping

**Review by nicosantangelo — COMMENTED** [review]
Awesome!

File: webapp/src/components/RankingsTable/RankingsTable.tsx [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+                onClick={() => setCurrentEntity(entity)}
+              >
+                {t(`home_page.analytics.rankings.${entity}.tab_title`)}
+              </Tabs.Tab>
+            ))}
+          </Tabs.Left>
+        </Tabs>
+        {currentEntity === RankingEntities.ITEMS && (
+          <>
+            <Dropdown
```
[the remark this answers is not in the document]
Comment (juanmahidalgo) [reply webapp/src/components/RankingsTable/RankingsTable.tsx:81]: lemme ask the product team, I don't think it looks good if I just the `search` prop 😛 
image

File: webapp/src/components/RankingsTable/RankingsTable.tsx [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+  useEffect(() => {
+    onFetchRankings(currentEntity, currentTimeframe, currentFilters)
+  }, [onFetchRankings, currentTimeframe, currentEntity, currentFilters])
+
+  const registerHandleFilterChange = (filterName: keyof RankingsFilters) => (
+    _event: React.SyntheticEvent<HTMLElement, Event>,
+    { value }: DropdownProps
+  ) => {
+    setCurrentFilters(prevValues => ({
+      ...prevValues,
```
[the remark this answers is not in the document]
Comment (juanmahidalgo) [reply webapp/src/components/RankingsTable/RankingsTable.tsx]: nope! updated!

File: webapp/src/components/RankingsTable/RankingsTable.tsx [hunk 5]
```diff
@@ -0,0 +1,431 @@
...
+        </Tabs>
+        {currentEntity === RankingEntities.ITEMS && (
+          <>
+            <Dropdown
+              defaultValue={ALL_FILTER}
+              direction="right"
+              options={[
+                ALL_FILTER,
+                ...Object.values(WearableCategory.schema.enum)
+              ].map(cat => ({
```
[the remark this answers is not in the document]
Comment (juanmahidalgo) [reply webapp/src/components/RankingsTable/RankingsTable.tsx]: updated!

File: webapp/src/components/RankingsTable/RankingsTable.tsx (hunk 1 above)
Comment (juanmahidalgo) [reply webapp/src/components/RankingsTable/RankingsTable.tsx]: nope! updated

File: webapp/src/components/RankingsTable/RankingsTable.tsx (hunk 2 above)
Comment (juanmahidalgo) [reply webapp/src/components/RankingsTable/RankingsTable.tsx]: totally right 😂 updated!

**Review by nicosantangelo — APPROVED** [review]

**Pull request merged**